themes/docs-new/assets/js/swiftype-config.js

```diff
@@ -0,0 +1,217 @@
+/*
+  Sources of information:
+  https://github.com/swiftype/swiftype-search-jquery
+  https://github.com/swiftype/swiftype-autocomplete-jquery
+  https://github.com/swiftype/swiftype-faceted-search-example
+  https://swiftype.com/documentation/site-search/guides/jquery
+  https://swiftype.com/documentation/site-search/guides/faceted-search
+*/
+
+$(document).ready(function() {
+
+  // Swiftype UI elements //
+
+  var showSearchModal = function() {
+    $("#swiftype-modal").show(500);
+    $("#swiftype-search-form-modal").focus();
+  }
+
+  var hideSearchModal = function() {
+    $("#swiftype-modal").hide(250);
+    $("#swiftype-custom-facets input:checkbox[class='product-filter']").prop( "checked", false );
+    window.location.hash = "";
+    $(".swiftype-search-input").val('');
+    $("div.swiftype-widget > div.autocomplete").html("<ul></ul>");
+  }
+
+  $("#swiftype-close-button").click(function(){
+    hideSearchModal();
+  });
+
+  $('body').click(function (event) {
+    if(!$(event.target).closest('#swiftype-modal-content').length && !$(event.target).is('#swiftype-modal-content')) {
+      hideSearchModal();
+    }
+  });
+
+  $("#swiftype-custom-facets-toggle-button").click(function(){
+    if ($(window).width() < 768) {
+      $("#swiftype-custom-facets").toggle(500);
+      $("#swiftype-custom-facets-toggle-caret").find('svg').toggle();
+    }
+  });
+
+  $('#swiftype-facet-infra').on('change',function(){
+    $('#swiftype-facet-client, #swiftype-facet-server').prop('checked',$(this).prop('checked'));
+  });
+
+  $(window).resize(function() {
+    if ($(this).width() < 768) {
+      $('#swiftype-custom-facets').hide();
+    } else {
+      $('#swiftype-custom-facets').show();
+      }
+  });
```

Comment:
I'd suggest this function get debounced or throttled.  This will help ease up this function getting called an unnecessary amount of times when a user is resizing the page.
https://bencentra.com/code/2015/02/27/optimizing-window-resize.html